bayes_opt/domain_reduction.py

```diff
@@ -96,46 +99,101 @@ def _update(self, target_space: TargetSpace) -> None:
+
+        Parameters:
+        -----------
+        new_bounds : np.array
+            The proposed new_bounds that (may) need adjustment.
+
+        global_bounds : np.array
+            The maximum allowable bounds for each parameter.
+
+        Returns:
+        --------
+        new_bounds : np.array
+            The adjusted bounds after enforcing constraints.
+        """
+
+        #sort bounds
         new_bounds = np.sort(new_bounds)
 
-        # check if the new bounds exceed the global bounds
+        # Validate each parameter's bounds against the global_bounds
         for i, pbounds in enumerate(new_bounds):
-            # check if lower bound exceeds range 
+            # If the lower bound of the parameter is outside the global bounds, reset the lower bound
             if (pbounds[0] < global_bounds[i, 0] or pbounds[0] > global_bounds[i, 1]):
                 pbounds[0] = global_bounds[i, 0]
-                warn("""Domain reduction warning:
-                    A parameter's lower bound has exceeded its global boundaries.
-                    The offensive boundary has been reset, but the optimizer may not converge.""")
-                
-            # check if upper bound exceeds range
+                warn("""Domain Reduction Warning:
+                    A parameter's lower bound has exceeded its global limit.
+                    The offensive boundary has been reset, but be cautious of optimizer convergence.""")
+
+            # If the upper bound bound of the parameter is outside the global bounds, reset the lower bound
             if (pbounds[1] > global_bounds[i, 1] or pbounds[1] < global_bounds[i, 0]):
                 pbounds[1] = global_bounds[i, 1]
                 warn("""Domain reduction warning:
-                    A parameter's upper bound has exceeded its global boundaries.
-                    The offensive boundary has been reset, but the optimizer may not converge.""")
-     
-        for i, entry in enumerate(new_bounds):
-            window_width = abs(entry[0] - entry[1])
-            if window_width < self.minimum_window[i]:
-                dw = (self.minimum_window[i] - window_width) / 2.0
-                left_expansion_space = abs(global_bounds[i, 0] - entry[0]) # should be non-positive
-                right_expansion_space = abs(global_bounds[i, 1] - entry[1]) # should be non-negative
-                # conservative
-                dw_l = min(dw, left_expansion_space)
-                dw_r = min(dw, right_expansion_space)
-                # this crawls towards the edge
-                ddw_r = dw_r + max(dw - dw_l, 0)
-                ddw_l = dw_l + max(dw - dw_r, 0)
-                new_bounds[i, 0] -= ddw_l
-                new_bounds[i, 1] += ddw_r
+                    A parameter's lower bound has exceeded its global limit.
+                    The offensive boundary has been reset, but be cautious of optimizer convergence.""")
```

Comment:
as above. Note that both of them state `lower bound has exceeded its global limit` but one of the cases presumably concerns the upper bound.